lib/ae_mdw_web/controllers/tx_controller.ex

```diff
@@ -77,8 +76,27 @@ defmodule AeMdwWeb.TxController do
   @doc """
   Counts each field for all transaction types where this address is present.
   """
-  def count_id(%Conn{assigns: %{state: state}} = conn, %{"id" => id}),
-    do: handle_input(conn, fn -> conn |> json(id_counts(state, Validate.id!(id))) end)
+  def count_id(%Conn{assigns: %{state: state}} = conn, %{"id" => id, "type_group" => group}) do
+    with {:ok, tx_type_group} <- Validate.tx_group(group),
+         {:ok, pubkey} <- Validate.id(id) do
+      handle_input(conn, fn ->
+        json(conn, count_id_type_group(state, pubkey, tx_type_group))
+      end)
+    end
+  end
+
+  def count_id(%Conn{assigns: %{state: state}} = conn, %{"id" => id, "type" => type}) do
+    with {:ok, tx_type} <- Validate.tx_type(type),
+         {:ok, pubkey} <- Validate.id(id) do
+      handle_input(conn, fn ->
+        json(conn, count_id_type(state, pubkey, tx_type))
+      end)
+    end
+  end
+
+  def count_id(%Conn{assigns: %{state: state}} = conn, %{"id" => id}) do
+    handle_input(conn, fn -> conn |> json(id_counts(state, Validate.id!(id))) end)
```

Comment:
Can we use `with {:ok, pubkey} <- Validate.id(id)` like we do on the other functions so that we can avoid using `handle_input/2` ?

Reply:
done